pkgs/tools/video/yamdi/default.nix

```diff
@@ -0,0 +1,31 @@
+{
+  stdenv,
+  fetchurl,
+}:
+
+stdenv.mkDerivation rec {
+  name = "yamdi-${version}";
+  version = "1.9";
+
+  # Source repo is also available here:
+  # https://github.com/ioppermann/yamdi
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/project/yamdi/yamdi/${version}/yamdi-${version}.tar.gz";
```

Comment:
Actually (as with many things in nixpkgs) there isn't really a consent on what's best and what to use when. Usually using source archives is better because downloading/unpacking/preparing them does only take a fraction of time compared to complete VCS checkouts. There are however cases when fetching from source repos might be preferred. For example if a package does not have a reliable release cycle, sometimes old package versions vanish from mirrors on new releases, mirrors with bad connectivity/down-times/security issues, some source archive do not contain all needed data (like git submodules) on the other hand source repos lack stuff sometimes too.

Here's a list of common package acquisition methods ordered least to most resource usage during builds:

* `fetchurl` (think `fetchStatic`) downloads the specified file and computes the hash from this file
* `fetchzip` (think `fetchDynamic`) downloads and unpacks most packed files (also tar.*, not only zip) but computes hashes from the unpacked data. Prefer this for dynamically created archives (like github) where compression settings or file creation/access timestamps might change but content does not.
* specialized VCS like `fetchFromGitHub`, unless it's using submodules, then use `fetchgit` instead
* generic fetchgit/hg/svn/cvs..